.github/actions/export-apis-data/index.js

```diff
@@ -0,0 +1,36 @@
+const fs = require('fs-extra');
+const path = require('path');
+const yaml = require('js-yaml');
+
+//joining path of directory
+const directoryPath = path.join(__dirname, '../../../catalogue');
+
+function yamlToJson(filePaths) {
+
+    let data = [];
+    let outputfile = 'apis.json';
+
+    filePaths.forEach(file => {
+        obj = yaml.load(fs.readFileSync(file, {encoding: 'utf-8'}));
+        data.push(obj);
+    });
+
+    // this code if you want to save
+    fs.writeFileSync(outputfile, JSON.stringify(data, null, 2))
+}
+
+const { readdir } = require('fs').promises;
+async function getFiles(dir) {
+  const dirents = await readdir(dir, { withFileTypes: true });
+  const files = await Promise.all(dirents.map((dirent) => {
+    const res = path.resolve(dir, dirent.name);
+    return dirent.isDirectory() ? getFiles(res) : res;
+  }));
+  return Array.prototype.concat(...files);
+}
+
+console.log(directoryPath);
+
+getFiles(directoryPath)
+  .then(files => yamlToJson(files.filter(file => file.includes('\\api.yml'))))
```

Comment:
Your action is set to run on ubuntu but you're testing for a windows file path "\\api.yml"

This might be why the action appears to be producing an empty file